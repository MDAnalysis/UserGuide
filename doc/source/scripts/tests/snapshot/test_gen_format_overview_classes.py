from unittest.mock import patch

from gen_format_overview_classes import (
    FILE_TYPES,
    CoordinateReaders,
    FormatOverview,
    SphinxClasses,
)


def test_FILE_TYPES():
    assert FILE_TYPES.keys() >= {"DCD", "GRO", "XTC", "ITP"}


def test_FormatOverview(snapshot):
    with patch("builtins.open"):
        ov = FormatOverview()
    assert ov.table_writer.lines == snapshot


def test_CoordinateReaders(snapshot):
    with patch("builtins.open"):
        cr = CoordinateReaders()
    assert cr.table_writer.lines == snapshot


def test_SphinxClasses(snapshot):
    with patch("builtins.open"):
        sc = SphinxClasses("PDB")
    assert sc.table_writer.lines == snapshot
