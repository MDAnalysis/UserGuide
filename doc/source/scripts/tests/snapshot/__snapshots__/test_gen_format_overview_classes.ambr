# serializer version: 1
# name: test_CoordinateReaders
  list([
    list([
      ':ref:`ARC <TXYZ-format>`',
      'Tinker file',
      '',
      '',
    ]),
    list([
      ':ref:`CHEMFILES <CHEMFILES-format>`',
      'use readers from `chemfiles <https://chemfiles.org/>`_ library',
      '',
      '',
    ]),
    list([
      ':ref:`CONFIG <CONFIG-format>`',
      'DL_Poly CONFIG file',
      '✓',
      '',
    ]),
    list([
      ':ref:`COOR <COOR-format>`',
      'NAMD binary restart file',
      '',
      '',
    ]),
    list([
      ':ref:`CRD <CRD-format>`',
      'CHARMM CARD file',
      '',
      '',
    ]),
    list([
      ':ref:`CRDBOX <TRJ-format>`',
      'AMBER ASCII trajectories',
      '',
      '',
    ]),
    list([
      ':ref:`DATA <DATA-format>`',
      'LAMMPS data file',
      '✓',
      '',
    ]),
    list([
      ':ref:`DCD <DCD-format>`',
      'CHARMM, NAMD, or LAMMPS binary trajectory',
      '',
      '',
    ]),
    list([
      ':ref:`DMS <DMS-format>`',
      'DESRES Molecular Structure file',
      '✓',
      '',
    ]),
    list([
      ':ref:`ENT <PDB-format>`',
      'Standard PDB file',
      '',
      '',
    ]),
    list([
      ':ref:`FHIAIMS <IN-format>`',
      'FHI-aims input file',
      '✓',
      '',
    ]),
    list([
      ':ref:`GMS <GMS-format>`',
      'GAMESS file',
      '',
      '',
    ]),
    list([
      ':ref:`GRO <GRO-format>`',
      'GROMACS structure file',
      '✓',
      '',
    ]),
    list([
      ':ref:`GSD <GSD-format>`',
      'HOOMD GSD file',
      '',
      '',
    ]),
    list([
      ':ref:`H5MD <H5MD-format>`',
      '`H5MD <https://www.nongnu.org/h5md/index.html>`_ trajectory format',
      '',
      '',
    ]),
    list([
      ':ref:`HISTORY <HISTORY-format>`',
      'DL_Poly HISTORY file',
      '✓',
      '',
    ]),
    list([
      ':ref:`IN <IN-format>`',
      'FHI-aims input file',
      '✓',
      '',
    ]),
    list([
      ':ref:`INPCRD <INPCRD-format>`',
      'AMBER restart file',
      '',
      '',
    ]),
    list([
      ':ref:`LAMMPS <LAMMPS-format>`',
      'a LAMMPS DCD trajectory',
      '',
      '',
    ]),
    list([
      ':ref:`LAMMPSDUMP <LAMMPSDUMP-format>`',
      'LAMMPS ascii dump file',
      '',
      '',
    ]),
    list([
      ':ref:`MDCRD <TRJ-format>`',
      'AMBER ASCII trajectories',
      '',
      '',
    ]),
    list([
      ':ref:`MMTF <MMTF-format>`',
      'MMTF file',
      '',
      '',
    ]),
    list([
      ':ref:`MOL2 <MOL2-format>`',
      'Tripos MOL2 file',
      '',
      '',
    ]),
    list([
      ':ref:`NAMDBIN <COOR-format>`',
      'NAMD binary restart file',
      '',
      '',
    ]),
    list([
      ':ref:`NC <NCDF-format>`',
      'AMBER NETCDF format',
      '✓',
      '✓',
    ]),
    list([
      ':ref:`NCDF <NCDF-format>`',
      'AMBER NETCDF format',
      '✓',
      '✓',
    ]),
    list([
      ':ref:`OPENMMAPP <OPENMMAPP-format>`',
      '`OpenMM <https://openmm.org/>`_ Application layer objects',
      '',
      '',
    ]),
    list([
      ':ref:`OPENMMSIMULATION <OPENMMSIMULATION-format>`',
      '`OpenMM <https://openmm.org/>`_ Simulation objects',
      '✓',
      '✓',
    ]),
    list([
      ':ref:`PARMED <PARMED-format>`',
      '`ParmEd <http://parmed.github.io/ParmEd/html/index.html>`_ Structure',
      '',
      '',
    ]),
    list([
      ':ref:`PDB <PDB-format>`',
      'Standard PDB file',
      '',
      '',
    ]),
    list([
      ':ref:`PDBQT <PDBQT-format>`',
      'PDBQT file',
      '',
      '',
    ]),
    list([
      ':ref:`PQR <PQR-format>`',
      'PQR file',
      '',
      '',
    ]),
    list([
      ':ref:`RDKIT <RDKIT-format>`',
      '`RDKit <https://www.rdkit.org/>`_ Molecule',
      '',
      '',
    ]),
    list([
      ':ref:`RESTRT <INPCRD-format>`',
      'AMBER restart file',
      '',
      '',
    ]),
    list([
      ':ref:`TNG <TNG-format>`',
      'Trajectory Next Generation file',
      '✓',
      '✓',
    ]),
    list([
      ':ref:`TRC <TRC-format>`',
      'GROMOS11 trajectory file',
      '',
      '',
    ]),
    list([
      ':ref:`TRJ <TRJ-format>`',
      'AMBER ASCII trajectories',
      '',
      '',
    ]),
    list([
      ':ref:`TRR <TRR-format>`',
      'GROMACS TRR trajectory',
      '✓',
      '✓',
    ]),
    list([
      ':ref:`TRZ <TRZ-format>`',
      'IBIsCO or YASP binary trajectory',
      '✓',
      '',
    ]),
    list([
      ':ref:`TXYZ <TXYZ-format>`',
      'Tinker file',
      '',
      '',
    ]),
    list([
      ':ref:`XPDB <XPDB-format>`',
      'Extended PDB file',
      '',
      '',
    ]),
    list([
      ':ref:`XTC <XTC-format>`',
      'GROMACS compressed trajectory',
      '',
      '',
    ]),
    list([
      ':ref:`XYZ <XYZ-format>`',
      'XYZ file',
      '',
      '',
    ]),
  ])
# ---
# name: test_FormatOverview
  list([
    list([
      ':ref:`ARC <TXYZ-format>`',
      'Tinker file',
      '✓',
      '✓',
      '✓',
      '',
    ]),
    list([
      ':ref:`CHEMFILES <CHEMFILES-format>`',
      'use readers from `chemfiles <https://chemfiles.org/>`_ library',
      '',
      '✓',
      '✓',
      '✓',
    ]),
    list([
      ':ref:`CONFIG <CONFIG-format>`',
      'DL_Poly CONFIG file',
      '✓',
      '✓',
      '✓',
      '',
    ]),
    list([
      ':ref:`COOR <COOR-format>`',
      'NAMD binary restart file',
      '',
      '✓',
      '✓',
      '✓',
    ]),
    list([
      ':ref:`CRD <CRD-format>`',
      'CHARMM CARD file',
      '✓',
      '✓',
      '✓',
      '✓',
    ]),
    list([
      ':ref:`CRDBOX <TRJ-format>`',
      'AMBER ASCII trajectories',
      '',
      '✓',
      '✓',
      '',
    ]),
    list([
      ':ref:`DATA <DATA-format>`',
      'LAMMPS data file',
      '✓',
      '✓',
      '✓',
      '✓',
    ]),
    list([
      ':ref:`DCD <DCD-format>`',
      'CHARMM, NAMD, or LAMMPS binary trajectory',
      '',
      '✓',
      '✓',
      '✓',
    ]),
    list([
      ':ref:`DMS <DMS-format>`',
      'DESRES Molecular Structure file',
      '✓',
      '✓',
      '✓',
      '',
    ]),
    list([
      ':ref:`ENT <PDB-format>`',
      'Standard PDB file',
      '✓',
      '✓',
      '✓',
      '✓',
    ]),
    list([
      ':ref:`FHIAIMS <IN-format>`',
      'FHI-aims input file',
      '✓',
      '✓',
      '✓',
      '✓',
    ]),
    list([
      ':ref:`GMS <GMS-format>`',
      'GAMESS file',
      '✓',
      '✓',
      '✓',
      '',
    ]),
    list([
      ':ref:`GRO <GRO-format>`',
      'GROMACS structure file',
      '✓',
      '✓',
      '✓',
      '✓',
    ]),
    list([
      ':ref:`GSD <GSD-format>`',
      'HOOMD GSD file',
      '✓',
      '✓',
      '✓',
      '',
    ]),
    list([
      ':ref:`H5MD <H5MD-format>`',
      '`H5MD <https://www.nongnu.org/h5md/index.html>`_ trajectory format',
      '',
      '✓',
      '✓',
      '✓',
    ]),
    list([
      ':ref:`HISTORY <HISTORY-format>`',
      'DL_Poly HISTORY file',
      '✓',
      '✓',
      '✓',
      '',
    ]),
    list([
      ':ref:`IN <IN-format>`',
      'FHI-aims input file',
      '✓',
      '✓',
      '✓',
      '✓',
    ]),
    list([
      ':ref:`INPCRD <INPCRD-format>`',
      'AMBER restart file',
      '',
      '✓',
      '✓',
      '',
    ]),
    list([
      ':ref:`ITP <ITP-format>`',
      'GROMACS portable topology file',
      '✓',
      '',
      '✓',
      '',
    ]),
    list([
      ':ref:`LAMMPS <LAMMPS-format>`',
      'a LAMMPS DCD trajectory',
      '',
      '✓',
      '✓',
      '✓',
    ]),
    list([
      ':ref:`LAMMPSDUMP <LAMMPSDUMP-format>`',
      'LAMMPS ascii dump file',
      '✓',
      '✓',
      '✓',
      '',
    ]),
    list([
      ':ref:`MDCRD <TRJ-format>`',
      'AMBER ASCII trajectories',
      '',
      '✓',
      '✓',
      '',
    ]),
    list([
      ':ref:`MMTF <MMTF-format>`',
      'MMTF file',
      '✓',
      '✓',
      '✓',
      '',
    ]),
    list([
      ':ref:`MOL2 <MOL2-format>`',
      'Tripos MOL2 file',
      '✓',
      '✓',
      '✓',
      '✓',
    ]),
    list([
      ':ref:`NAMDBIN <COOR-format>`',
      'NAMD binary restart file',
      '',
      '✓',
      '✓',
      '✓',
    ]),
    list([
      ':ref:`NC <NCDF-format>`',
      'AMBER NETCDF format',
      '',
      '✓',
      '✓',
      '✓',
    ]),
    list([
      ':ref:`NCDF <NCDF-format>`',
      'AMBER NETCDF format',
      '',
      '✓',
      '✓',
      '✓',
    ]),
    list([
      ':ref:`OPENMMAPP <OPENMMAPP-format>`',
      '`OpenMM <https://openmm.org/>`_ Application layer objects',
      '✓',
      '✓',
      '✓',
      '',
    ]),
    list([
      ':ref:`OPENMMSIMULATION <OPENMMSIMULATION-format>`',
      '`OpenMM <https://openmm.org/>`_ Simulation objects',
      '',
      '✓',
      '✓',
      '',
    ]),
    list([
      ':ref:`OPENMMTOPOLOGY <OPENMMTOPOLOGY-format>`',
      '`OpenMM <https://openmm.org/>`_ Topology object',
      '✓',
      '',
      '✓',
      '',
    ]),
    list([
      ':ref:`PARM7 <TOP-format>`',
      'AMBER topology file',
      '✓',
      '',
      '✓',
      '',
    ]),
    list([
      ':ref:`PARMED <PARMED-format>`',
      '`ParmEd <http://parmed.github.io/ParmEd/html/index.html>`_ Structure',
      '✓',
      '✓',
      '✓',
      '✓',
    ]),
    list([
      ':ref:`PDB <PDB-format>`',
      'Standard PDB file',
      '✓',
      '✓',
      '✓',
      '✓',
    ]),
    list([
      ':ref:`PDBQT <PDBQT-format>`',
      'PDBQT file',
      '✓',
      '✓',
      '✓',
      '✓',
    ]),
    list([
      ':ref:`PQR <PQR-format>`',
      'PQR file',
      '✓',
      '✓',
      '✓',
      '✓',
    ]),
    list([
      ':ref:`PRMTOP <TOP-format>`',
      'AMBER topology file',
      '✓',
      '',
      '✓',
      '',
    ]),
    list([
      ':ref:`PSF <PSF-format>`',
      'CHARMM, NAMD, or XPLOR PSF file',
      '✓',
      '',
      '✓',
      '',
    ]),
    list([
      ':ref:`RDKIT <RDKIT-format>`',
      '`RDKit <https://www.rdkit.org/>`_ Molecule',
      '✓',
      '✓',
      '✓',
      '✓',
    ]),
    list([
      ':ref:`RESTRT <INPCRD-format>`',
      'AMBER restart file',
      '',
      '✓',
      '✓',
      '',
    ]),
    list([
      ':ref:`TNG <TNG-format>`',
      'Trajectory Next Generation file',
      '',
      '✓',
      '✓',
      '',
    ]),
    list([
      ':ref:`TOP <TOP-format>`',
      'AMBER topology file',
      '✓',
      '',
      '✓',
      '',
    ]),
    list([
      ':ref:`TPR <TPR-format>`',
      'GROMACS run topology file',
      '✓',
      '',
      '✓',
      '',
    ]),
    list([
      ':ref:`TRC <TRC-format>`',
      'GROMOS11 trajectory file',
      '',
      '✓',
      '✓',
      '',
    ]),
    list([
      ':ref:`TRJ <TRJ-format>`',
      'AMBER ASCII trajectories',
      '',
      '✓',
      '✓',
      '',
    ]),
    list([
      ':ref:`TRR <TRR-format>`',
      'GROMACS TRR trajectory',
      '',
      '✓',
      '✓',
      '✓',
    ]),
    list([
      ':ref:`TRZ <TRZ-format>`',
      'IBIsCO or YASP binary trajectory',
      '',
      '✓',
      '✓',
      '✓',
    ]),
    list([
      ':ref:`TXYZ <TXYZ-format>`',
      'Tinker file',
      '✓',
      '✓',
      '✓',
      '',
    ]),
    list([
      ':ref:`XML <XML-format>`',
      'HOOMD XML file',
      '✓',
      '',
      '✓',
      '',
    ]),
    list([
      ':ref:`XPDB <XPDB-format>`',
      'Extended PDB file',
      '✓',
      '✓',
      '✓',
      '',
    ]),
    list([
      ':ref:`XTC <XTC-format>`',
      'GROMACS compressed trajectory',
      '',
      '✓',
      '✓',
      '✓',
    ]),
    list([
      ':ref:`XYZ <XYZ-format>`',
      'XYZ file',
      '✓',
      '✓',
      '✓',
      '✓',
    ]),
  ])
# ---
# name: test_SphinxClasses
  list([
    list([
      '**Coordinate reader**',
      ':class:`MDAnalysis.coordinates.PDB.PDBReader`',
    ]),
    list([
      '**Coordinate writer**',
      ':class:`MDAnalysis.coordinates.PDB.PDBWriter`',
    ]),
    list([
      '**Topology parser**',
      ':class:`MDAnalysis.topology.PDBParser.PDBParser`',
    ]),
  ])
# ---
